SharedSources/VLCMetadata.m

```diff
@@ -53,19 +63,54 @@ - (void)updateMetadataFromMediaPlayer:(VLCMediaPlayer *)mediaPlayer;
         /* MLKit knows better than us if this thing is audio only or not */
         self.isAudioOnly = [item isSupportedAudioFile];
     } else {
-#endif
-        NSDictionary * metaDict = mediaPlayer.media.metaDictionary;
+        [self fillFromMetaDict:mediaPlayer];
+    }
+
+    [self checkIsAudioOnly:mediaPlayer];
 
-        if (metaDict) {
-            self.title = metaDict[VLCMetaInformationNowPlaying] ? metaDict[VLCMetaInformationNowPlaying] : metaDict[VLCMetaInformationTitle];
-            self.artist = metaDict[VLCMetaInformationArtist];
-            self.albumName = metaDict[VLCMetaInformationAlbum];
-            self.trackNumber = metaDict[VLCMetaInformationTrackNumber];
+    if (self.isAudioOnly) {
+        self.artworkImage = [VLCThumbnailsCache thumbnailForManagedObject:item];
+
+        if (self.artworkImage) {
+            if (self.artist)
+                self.title = [self.title stringByAppendingFormat:@" — %@", self.artist];
+            if (self.albumName)
+                self.title = [self.title stringByAppendingFormat:@" — %@", self.albumName];
         }
```

Comment:
Consider caching the metadata variables to save 1 selector dispatch per property.